### ISSUE TITLE
mdInput not apply custom theme

### ISSUE THREAD

**Issue by pmd30011991** [angular/material2#3470]
#### Bug, feature request, or proposal:
Hi team, this is my scss file to change theme, some components don't apply my custom theme such as "mdInput" when I hover on it, still using the default theme scheme ( blue color )
`
@import '../../../node_modules/@angular/material/core/theming/all-theme';
@include mat-core();

  $dark-primary: mat-palette($mat-light-green,500,800);
  $dark-accent:  mat-palette($mat-amber, A200, A100, A400);
  $dark-warn:    mat-palette($mat-deep-orange);

  $dark-theme: mat-light-theme($dark-primary, $dark-accent, $dark-warn);

  @include angular-material-theme($dark-theme);
`

Please help. Thanks

**Comment by thibaultsavary** [comment]
Questions are to ask there to avoid the team loosing time on non-bug / feature related issues.
https://gitter.im/angular/material2

IF you want the theme to be dark you have to use the `mat-dark-theme` function instead of the `mat-light-theme` function.

Your code seems OK to me.

**Be sure that the theme is defined inside `./src/style.scss` and not inside a component.**
I have a very similar code, and it's working correctly.

```scss
@import '~@angular/material/core/theming/all-theme';
@include mat-core();
$primary: mat-palette($mat-pink, 500, 700);
$accent: mat-palette($mat-indigo, A200, A100, A400);
$warn: mat-palette($mat-deep-orange);
$grey: mat-palette($mat-grey);
$theme: mat-light-theme($primary, $accent, $warn);
@include angular-material-theme($theme);
```

**Comment by kara** [comment]
@pmd30011991  You're not still using the prebuilt theme file somewhere, are you?

**Comment by JoshMarler** [comment]
Also experiencing this issue since updating to latest master branch this morning. The md-placeholder does not appear to be styled with &primary when focused/entered/. My theme is not being applied. The mdInput underline however is styled with the appropriate &primary color from my theme. 

It's just the placeholder/floating label that appears to be an issue. Will investigate further.

**Comment by JoshMarler** [comment]
I can force this to work using the following: 

```
.blue-color-scheme {
  $primary: mat-palette($mat-blue);
  $accent: mat-palette($mat-blue-grey);
  $blue-theme: mat-light-theme($primary, $accent);

  @include custom-sidenav($primary);
  @include angular-material-theme($blue-theme);

  md-input-container.mat-focused {
    .mat-input-placeholder {
      color: mat-color($primary) !important;
    }
  }
}
```

Apologies, my scss and material knowledge isn't good enough to know why this issue has suddenly occurred. I need to use the latest branch so that I can set the theme for my apps OverlayContainer so dialogs and the like are correctly styled.

```
 constructor(overlayContainer: OverlayContainer) {
    overlayContainer.themeClass = 'blue-color-scheme';
  }
```

**Comment by pmd30011991** [comment]
Hi thanks for you efforts

The issue to be resolved if we remove the default theme located inside <header> by <link> tag. if I not wrong it's " indigo pink" 
I guess there are some overwritten if we link directly the css into the page, so custom style will not apply to some parts.

Regards 

Duy Phung
Email : pmd.30011991@gmail.com
Phone: (+84)933649722 - (+60)1121526134
Skype : pmd30011991

**Comment by JoshMarler** [comment]
@pmd30011991 

Hmm, interesting. I don't actually have a reference to one of the default themes anywhere in my app.

**Comment by dannyweb8** [comment]
@kara : I still keep using the pre-build theme file and it link directly to the html file.

**Comment by JoshMarler** [comment]
Incidentally this issue only occurs when my app content is contained within an md-sidenav-container. Will keep playing.

**Comment by JoshMarler** [comment]
My themes are currently laid out as follows in my styles.scss:


```
@import "../node_modules/@angular/material/core/theming/all-theme";

@include mat-core();

$primary: mat-palette($mat-green);
$accent: mat-palette($mat-light-green);
$warn: mat-palette($mat-red);
$green-theme: mat-light-theme($primary, $accent, $warn);

@include angular-material-theme($green-theme);


.blue-color-scheme {
  $primary: mat-palette($mat-blue);
  $accent: mat-palette($mat-blue-grey);
  $warn: mat-palette($mat-red);
  $blue-theme: mat-light-theme($primary, $accent, $warn);

  @include angular-material-theme($blue-theme);
}

.yellow-color-scheme {
  $primary: mat-palette($mat-yellow);
  $accent: mat-palette($mat-indigo);
  $warn: mat-palette($mat-red);
  $yellow-theme: mat-light-theme($primary, $accent, $warn);

  @include angular-material-theme($yellow-theme);
}
```

I'm using themes gated by different selectors to allow different app color schemes for different customers at the app bootstrapping stage.

I must be missing something here..

**Comment by jelbourn** [comment]
Closing this issue as it does not follow the issue template. Please file a new issue that follows the template, which gives the team the information needed to investigate.

**Comment by shawinder** [comment]
`overlaycontainer` and `FullscreenOverlayContainer` are throwing error in latest angular material saying `no such exported member found`. Please let me know the alternative modules?
